### ISSUE TITLE
[bug]: It gets stuck at loading when trying to open a sub-category that has been redirected

### ISSUE THREAD

**Issue by truongnx-dtn** [magento/pwa-studio#2860]
**Describe the bug**
It gets stuck at loading when trying to open a subcategory that has been redirected.
In the source code, it used history.replace() function but missing the "/" for params.

**To reproduce**
1. Login to admin panel
2. Go to "Marketing / SEO & Search / URL Rewrites"
3. Add an URL rewrite for a subcategory (use 301 or 302 redirect)
4. Go to front-page and clear the browser cache
5. Navigate to the subcategory
6. See error

**Expected behavior**
It should redirect to rewritten URL.

**Screenshots**
![image](https://user-images.githubusercontent.com/66080981/98905474-0df66500-24ee-11eb-929c-cb962c478d6e.png)

![image](https://user-images.githubusercontent.com/66080981/98906404-9295b300-24ef-11eb-980b-7c772da3d5c9.png)

**Possible solutions**
Append "/" to the head of params will fix the error.

**Device information**
 - Device [PC]:
 - OS [Ubuntu 20.04, Windows 10]:
 - Browser [Chrome]:
 - Browser Version [86]:

**Packages this bug is in regards to**
- [ ] `venia-concept`
- [ ] `venia-ui`
- [ ] `pwa-buildpack`
- [x] `peregrine`
- [ ] `pwa-devdocs`
- [ ] `upward-js`
- [ ] `upward-spec`
- [ ] `create-pwa`

**Comment by truongnx-dtn** [comment]
@magento I am working on this

**Comment by truongnx-dtn** [comment]
@magento add to contributors team

**Comment by chiranjeevi-cj** [comment]
@magento i am working on this

**Comment by chiranjeevi-cj** [comment]
@truongnx-dtn Below added screenshot is that what needs to be added?
![Screenshot from 2020-11-11 19-52-48](https://user-images.githubusercontent.com/52098385/98823117-b6a5b580-2457-11eb-99cc-fc5e63fd1e21.png)

**Comment by truongnx-dtn** [comment]
@magento i am working on this

**Comment by truongnx-dtn** [comment]
> @truongnx-dtn Below added screenshot is that what needs to be added?
> ![Screenshot from 2020-11-11 19-52-48](https://user-images.githubusercontent.com/52098385/98823117-b6a5b580-2457-11eb-99cc-fc5e63fd1e21.png)

No. Sorry for unclear description. I've a pull request here, but how can I link it to this issue
https://github.com/magento/pwa-studio/pull/2861

**Comment by truongnx-dtn** [comment]
Hi @chiranjeevi-cj, sorry for my unclear description.
Do you know how to add a version label to my pull request? I think I don't have permission.

**Comment by chiranjeevi-cj** [comment]
@magento i am working on this

**Comment by chiranjeevi-cj** [comment]
@truongnx-dtn that will be added by the maintainer(reviewers). We are contributors, we don't have access to do.

**Comment by truongnx-dtn** [comment]
@chiranjeevi-cj Many thanks!
I have one more question: we must call maintainer to review or just wait?

**Comment by sirugh** [comment]
@magento export issue to JIRA project PWA as Bug

**Comment by sirugh** [comment]
@magento export issue to JIRA project PWA as Story

**Comment by magento-engcom-team** [comment]
:white_check_mark: Jira issue (https://jira.corp.magento.com/browse/PWA-1279) is successfully created for this issue.

**Comment by vasilii-b** [comment]
Couldn't reproduce the issue on the latest `develop` branch.

![Storefront](https://user-images.githubusercontent.com/13456702/108372633-b0b63200-7207-11eb-8b06-06f9d2efba27.png)

![Admin-URL-Rewrites-Entries](https://user-images.githubusercontent.com/13456702/108372673-b875d680-7207-11eb-8e3e-05574437245b.png)

Here is what I did:

1. Took a category for the experiment `women/top-women.html`
2. Removed the url rewrite and created new one `top-women.html` -> `catalog/category/view/id/21`
3. Created another url rewrite `women/top-women.html` -> `top-women.html` with redirect type 301 (permanent)
